compiler/transforms/stream_snaxify.py

```diff
@@ -0,0 +1,161 @@
+from dataclasses import dataclass
+
+from xdsl.dialects import builtin, memref, memref_stream
+from xdsl.dialects.builtin import MemRefType, ModuleOp, StringAttr
+from xdsl.ir import MLContext
+from xdsl.ir.affine import AffineMap
+from xdsl.passes import ModulePass
+from xdsl.pattern_rewriter import (
+    PatternRewriter,
+    PatternRewriteWalker,
+    RewritePattern,
+    op_type_rewrite_pattern,
+)
+from xdsl.traits import SymbolTable
+
+from compiler.dialects import snax_stream
+from compiler.dialects.accfg import AcceleratorOp
+from compiler.dialects.snax import StreamerConfigurationAttr
+
+
+@dataclass
+class MemrefStreamToSnaxPattern(RewritePattern):
+    """
+    A pass to convert memref_stream operations to snax stream.
+
+    This boils down to combining the data access patterns of a memref_stream op (operation -> data),
+    with a certain data layout: an affine map from (data -> memory) into a mapping (operation -> memory).
+
+    This takes the form of a snax_stream access pattern, mapping (operation -> memory)
+    which, in hardware, is  realized by the Streamers.
+
+    Current restrictions:
+        We are only handling default memory layouts for now (NoneAttr)
+    """
+
+    @op_type_rewrite_pattern
+    def match_and_rewrite(
+        self, op: memref_stream.StreamingRegionOp, rewriter: PatternRewriter
+    ):
+        # Compliance checks:
+
+        # Handle only memref stream ops dispatched to an accelerator:
+        if "accelerator" not in op.attributes:
+            return
+
+        # Go and fetch the accelerator op
+        accelerator_str = op.attributes["accelerator"]
+        assert isinstance(accelerator_str, StringAttr)
+
+        module_op = op
+        while module_op and not isinstance(module_op, ModuleOp):
+            module_op = module_op.parent_op()
+        if not module_op:
+            raise RuntimeError("Module Op not found")
+
+        trait = module_op.get_trait(SymbolTable)
+        assert trait is not None
+        acc_op = trait.lookup_symbol(module_op, accelerator_str)
+
+        if not isinstance(acc_op, AcceleratorOp):
+            raise RuntimeError("AcceleratorOp not found!")
+
+        streamer_config = acc_op.attributes["streamer_config"]
+        if not isinstance(streamer_config, StreamerConfigurationAttr):
+            raise RuntimeError("Streamer interface not found for given accelerator op")
```

Comment:
I think one centralized helper for this would also be nice, since it was also in your other PR.